src/main/java/com/labsynch/labseer/service/BulkLoadServiceImpl.java

```diff
@@ -463,6 +469,18 @@ public BulkLoadRegisterSDFResponseDTO registerSdf(BulkLoadRegisterSDFRequestDTO
 				if (validate) {
 					try {
 						parent = validateParentAgainstDryRunCompound(parent, numRecordsRead, results);
+						// Check list of aliases within file being bulkloaded
+						if (!propertiesUtilService.getAllowDuplicateParentAliases()) {
+							for (ParentAlias alias : parent.getParentAliases()) {
+								if (allAliasMaps.get(alias.getAliasName()) == null) {
+									allAliasMaps.put(alias.getAliasName(), numRecordsRead);
+								} else {
+									throw new NonUniqueAliasException(
+											"Within File, Parent alias " + alias.getAliasName()
```

Comment:
nit: capitalization. I'd prefer `Within file, Parent Alias xxx is not unique`

Reply:
Thanks, fixed.